types/trusted-types/index.d.ts

```diff
@@ -0,0 +1,42 @@
+// Type definitions for trusted-types 1.0
+// Project: https://github.com/WICG/trusted-types
+// Definitions by: Jakub Vrana <https://github.com/vrana>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare class TrustedHTML {
+	private readonly _TrustedHTMLBrand: true; // To prevent structural typing.
+}
+
+declare class TrustedScript {
+	private readonly _TrustedScriptBrand: true; // To prevent structural typing.
+}
+
+declare class TrustedScriptURL {
```

Comment:
Should these extend from each other? The original source code does, and it might make `instanceof` work better. I'm not familiar with how this library is supposed to be used though.

Reply:
They should not. The goal of this declaration is to describe the spec (to be published soon) with one working implementation (`google-chrome-unstable --enable-blink-features=TrustedDOMTypes --enable-experimental-web-platform-features`). They don't extend from each other there. https://github.com/WICG/trusted-types is just a polyfill and I will fix that instead.

Reply:
Spec: https://wicg.github.io/trusted-types/dist/spec/